source/extensions/filters/network/common/redis/client_impl.cc

```diff
@@ -49,9 +50,10 @@ ConfigImpl::ConfigImpl(
 ClientPtr ClientImpl::create(Upstream::HostConstSharedPtr host, Event::Dispatcher& dispatcher,
                              EncoderPtr&& encoder, DecoderFactory& decoder_factory,
                              const Config& config) {
-
-  std::unique_ptr<ClientImpl> client(
-      new ClientImpl(host, dispatcher, std::move(encoder), decoder_factory, config));
+  auto redis_command_stats = std::make_shared<RedisCommandStats>(
```

Comment:
I think we want to hold RedisCommandStats in your Config, not in ClientImpl. It will be expensive to create, basically read-only, and there could be more than one Redis client in a process.

Reply:
I will give this a try tmw.